### ISSUE TITLE
How to inject a React component in the Markdown AST?

### ISSUE THREAD

**Issue by jojolatina** [gatsbyjs/gatsby#20318]
## Summary
I'm writing a transformer plugin that converts Markdown images into some custom HTML for a better UX. Currently, it visits the MDAST for `image` tags and transforms it into HTML:
```js
module.exports = ({ markdownAST }) => {
  visit(markdownAST, 'image', node => {
      const html = `...`
      node.type = 'html'
      node.children = undefined
      node.value = html
  })

  return markdownAST
}
```
Now the hardcoded HTML comes to its limits and I'd like to use React components. So how can I inject a React component at this point? I tried to import React but it doesn't work at all.

Maybe I go the wrong way with my transformer plugin? The goal is to be able to write standard Markdown images:
```md
...
![](image.png)
...
```

And that this image gets converted to a React component in the final HTML. Using `gatsby-plugin-mdx` is not a solution, I want to stick with standard Markdown.

To summarize, my problem is just that instead of hardcoding some html in the transformer plugin, I want to use React components.

**Comment by jojolatina** [comment]
I figured out how to do this, here's how:

In my markdown, I write my regular image:
```md
![](image.png)
```

Then my transformer plugin visits the images and replaces them by this html:
```html
<myCustomImage src="${url}"></myCustomImage>
```

And `rehype-react` renders the htmlAST in the article template
